.gitignore

```diff
@@ -1,5 +1,5 @@
 /bin
-/build
+/build*
```

Comment:
This is a function of our workflows - separate test directories for build_debug, build_release etc. If this is a huge problem, I can remove it. If it's alright to keep, it would simplify our uses of this repo

Reply:
Oh I see.  Our convention it to use nested build trees under the `out/` directory.  See the top level Makefile for how that work.  e.g. the default build if you just type `make` will use cmake to create a build directory in `out/clang/Debug/`.   But if you use gcc it will go in `out/gcc/Debug/`, etc.  

I'm not sure why `/build` is in this file at all.. perhaps we should remove it.  Unless maybe its the default directory used by some higher level tool for dealing with cmake projects maybe?

Can you configure your tool to use `out/` ?  If not I'm OK landing this change, but maybe as a separate change on it own?